File: LC-412/LC-412.cpp
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    /**
     * Concepts:
     * Naive Approach
     */
    vector<string> fizzBuzz(int n) {
        vector<string> result;
        for (int i = 1; i <= n; ++i) {
            cout << i << endl;
            string temp = "";
            if (i % 3 == 0) {
                temp += "Fizz";
            }
            if (i % 5 == 0) {
                temp += "Buzz";
            }
            result.push_back(temp != "" ? temp : to_string(i));
        }
        return result;
    }
};

File: LC-412/LC-412.js
/**
 * @param {number} n
 * @return {string[]}
 */
var fizzBuzz = function(n) {
  /**
   * Concepts:
   * Optimize naive approach
   */

  let result = [];
  for (let i = 1; i <= n; ++i) {
    let temp = '';
    if (i % 3 === 0) {
      temp += 'Fizz';
    }
    if (i % 5 === 0) {
      temp += 'Buzz';
    }
    result.push(temp ? temp : String(i));
  }
  return result;
};
